app/services/oauth_clients/create_service.rb

```diff
@@ -36,10 +36,9 @@ module OAuthClients
   class CreateService < ::BaseServices::Create
     protected
 
-    def before_perform(params, _service_result)
-      OAuthClient.where(integration: params[:integration]).delete_all
-
-      super
+    def after_validate(params, contract_call)
```

Comment:
❓ why changing the hook? what is the difference, in the `before_perform` and the `after_validate`?

Reply:
Sure, good eye- the difference is the order of operations: if the contract fails validation, we do not want to purge the database of the previous OAuth clients i.e. we should only purge the previous oauth clients, only with a valid contract.

Ref: https://github.com/opf/openproject/blob/dev/app/services/base_services/base_contracted.rb#L59-L68

Reply:
makes perfect sense, question answered